### PR TITLE
Cleanup summary file checker

### PR DESCRIPTION
**Pull request by white238**
First step to ATS work is to have a test checker.

I am open to better names ('field', 'sample', etc)..

After names are settled on, I will review places I used it in the code.  Like here: https://github.com/LLNL/serac/blob/8a913aae29ac92ce9892d9bd1cfd2b0166c42e62/src/serac/physics/base_physics.cpp#L197

Example of summary file:

```
{
  "user_name": "white238",
  "host_name": "rzgenie8",
  "mpi_rank_count": 1,
  "curves": 
  {
    "t": [0.25, 0.5, 0.75, 1.0],
    "temperature": 
    {
      "l1norms": [10.8455612450063, 10.8769266938811, 10.9262526637677, 10.9890569744431],
      "l2norms": [4.00325571439491, 3.98778843947879, 3.988193145272, 3.99834571962069],
      "linfnorms": [1.99858754555508, 1.99602762948269, 1.99349091868975, 1.9915520450815],
      "avgs": [1.36394528852752, 1.36810524509546, 1.374153349967, 1.38184017005875],
      "mins": [1.00067514812945, 1.00311180843846, 1.00816133998986, 1.01610881694094],
      "maxs": [2.0, 2.0, 2.0, 2.0]
    },
    "velocity": 
    {
      "l1norms": [0.0, 0.0, 0.0, 0.0],
      "l2norms": [0.0, 0.0, 0.0, 0.0],
      "linfnorms": [0.0, 0.0, 0.0, 0.0],
      "avgs": [0.0, 0.0, 0.0, 0.0],
      "mins": [0.0, 0.0, 0.0, 0.0],
      "maxs": [0.0, 0.0, 0.0, 0.0]
    },
    "displacement": 
    {
      "l1norms": [0.0, 0.924962166264643, 1.84283834206585, 2.74688270159081],
      "l2norms": [0.0, 0.4275976012174, 0.851821419489431, 1.26946055313981],
      "linfnorms": [0.0, 0.310353229084647, 0.618487529803454, 0.921952178076032],
      "avgs": [0.0, 0.0391142766414546, 0.0763267241134098, 0.111371682619804],
      "mins": [0.0, -0.0382275916996528, -0.0908270569236879, -0.156722895441363],
      "maxs": [0.0, 0.317086042420179, 0.632902614360887, 0.944383108329756]
    }
  },
  "sidre_group_name": "serac_summary"
}
```

### CODE REVIEW

File: src/serac/physics/base_physics.cpp [hunk 1]
```diff
@@ -188,7 +188,7 @@ void BasePhysics::initializeSummary(axom::sidre::DataStore& datastore, double t_
 
   // t: array of each time step value
   axom::sidre::View*         t_array_view = curves_group->createView("t");
-  axom::sidre::Array<double> ts(t_array_view, array_size);
+  axom::sidre::Array<double> ts(t_array_view, 0, array_size);
```
Comment (white238) [inline src/serac/physics/base_physics.cpp:191]: I messed this up in my previous PR.  The constructor I missed that I was using was this:

```
array(sidre_view, num_elements, capacity=0);
```

File: scripts/testing/tolerance_test.py [hunk 2]
```diff
@@ -1,78 +1,221 @@
...
+    if len(baseline_field_names) > len(test_field_names):
+        print("ERROR: Test file is missing field names that are in the baseline file")
+        missing = list_missing(baseline_field_names, test_field_names)
+        print("       Missing field names: {0}".format(", ".join(missing)))
+        error_found = True
+    elif len(baseline_field_names) < len(test_field_names):
+        print("ERROR: Test file has extra field names not in baseline file")
+        missing = list_missing(test_field_names, baseline_field_names)
+        print("       Extra field names: {0}".format(", ".join(missing)))
+        error_found = True
```
Comment (samuelpmishLLNL) [inline scripts/testing/tolerance_test.py]: this was the construction I was referring to in the meeting, it looks like 
```python
test_names = ["displacement", "velocity"]
baseline_names = ["temperature", "voltage"]
```
might just fall through these conditionals, despite there being differences between the two lists. Another implementation might be something like

```python
if (test_names != baseline_names):
    print("ERROR: field names are different than baseline names")
    print(f"names present in test, but not baseline: {set(test_names).difference(set(baseline_names))}")
    print(f"names present in baseline, but not test: {set(baseline_names).difference(set(test_names))}")
```

File: scripts/testing/tolerance_test.py [hunk 3]
```diff
@@ -1,78 +1,221 @@
...
+import os
+import sys
+
+
+# This script compares two Serac summary files against each other.
+# The 'baseline' file is treated as a truth against the possibly wrong
+# 'test' file. It attempts to fail with the most error messages it can at a given
+# correctness level (field names > sample names > sample values).
+# Summary files were the same within the given tolerance if the
+# script exists successfully and not otherwise.
```
Comment (samuelpmishLLNL) [inline scripts/testing/tolerance_test.py]: regarding alternate names: I think of the {norms, mean, min, max} as "stats" / "statistics", "features", or "properties" of a given field

**Comment by white238** [comment]
Is it safe to assume we should also check that the timesteps, `t`, match?

File: scripts/testing/tolerance_test.py (hunk 3 above)
Comment (jamiebramwell) [reply scripts/testing/tolerance_test.py]: I like "stats" or "metrics".

File: scripts/testing/tolerance_test.py [hunk 4]
```diff
@@ -1,78 +1,235 @@
...
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Compare two Serac summary files")
+
+    parser.add_argument("--baseline", type=str, required=True,
+                        help="Path to baseline summary file")
+    parser.add_argument("--test", type=str, required=True,
+                        help="Path to test summary file")
+    parser.add_argument("--tolerance", type=float, required=True,
```
Comment (jamiebramwell) [inline scripts/testing/tolerance_test.py:40]: It would be nice if we could specify this on a per-field basis, but that could be a follow-on PR.

**Review by jamiebramwell — APPROVED** [review]

File: scripts/testing/tolerance_test.py (hunk 4 above)
Comment (white238) [reply scripts/testing/tolerance_test.py:40]: #601

**Review by samuelpmishLLNL — APPROVED** [review]

**Comment by samuelpmishLLNL** [comment]
Out of curiosity, do these summary files exist purely for automated correctness-testing, or do we expect developers/users to be opening them up to check that values are in the expected ranges? If a human needs to be able to read them, we might consider adopting a different format, since this one will become difficult to read without external tools when there are more than a few timesteps present. 

For example, what was happening around time `t = 4000` given the data below (using 250 timesteps as an example):

```json
{
"t": [0,1.000000000000000,3.116658318641999,6.060086981831167,9.713559075160374,14.00589309483133,18.88722050361819,24.31984404491886,30.27384469521906,36.72465422735958,43.65158322401658,51.03686666650779,58.86501289862737,67.12234146015290,75.79664425067244,84.87693041290635,94.35322990663045,104.2164393606721,114.4581990969513,125.0707936057052,136.0470699770248,147.3803702967770,159.0644750535941,171.0935553362856,183.4621321274756,196.1650413838438,209.1974038785143,222.5545989954727,236.2322418290065,250.2261630667030,264.5323912321827,279.1471369404186,294.0667788792409,309.2878512791569,324.8070326726881,340.6211357761038,356.7270983522955,373.1219749347801,389.8029293103777,406.7672276726990,424.0122323707647,441.5353961873142,459.3342570899883,477.4064334058845,495.7496193762067,514.3615810530437,533.2401525048697,552.3832323012845,571.7887802508943,591.4548143691753,611.3794080557088,631.5606874624061,651.9968290362816,672.6860572220446,693.6266423112744,714.8168984262685,736.2551816278217,757.9398881372271,779.8694526637094,802.0423468293192,824.4570776840420,847.1121863045313,870.0062464704530,893.1378634129552,916.5056726302433,940.1083387656658,963.9445545440939,988.0130397627258,1012.312540332757,1036.841827368635,1061.599696321880,1086.584966156684,1111.796478564704,1137.233097216674,1162.893707048614,1188.777213580604,1214.882542266208,1241.208637870787,1267.754463877061,1294.519001916383,1321.501251224308,1348.700228119120,1376.114965502084,1403.744512378252,1431.587933396755,1459.644308409545,1487.912732047658,1516.392313314090,1545.082175192460,1573.981454270668,1603.089300378819,1632.404876240710,1661.927357138230,1691.655930588073,1721.589796030154,1751.728164527221,1782.070258475112,1812.615311323201,1843.362567304550,1874.311281175353,1905.460717963246,1936.810152724107,1968.358870306968,2000.106165126698,2032.051340944132,2064.193710653313,2096.532596075574,2129.067327760163,2161.797244791152,2194.721694600367,2227.840032786103,2261.151622937396,2294.655836463627,2328.352052429254,2362.239657393475,2396.318045254623,2430.586617099133,2465.044781054878,2499.691952148749,2534.527552168288,2569.551009527243,2604.761759134894,2640.159242269028,2675.742906452409,2711.512205332639,2747.466598565279,2783.605551700128,2819.928536070531,2856.435028685629,2893.124512125449,2929.996474438723,2967.050409043367,3004.285814629507,3041.702195064986,3079.299059303268,3117.075921293646,3155.032299893702,3193.167718783922,3231.481706384427,3269.973795773719,3308.643524609407,3347.490435050835,3386.514073683564,3425.713991445632,3465.089743555567,3504.640889442073,3544.366992675352,3584.267620900009,3624.342345769494,3664.590742882042,3705.012391718046,3745.606875578854,3786.373781526917,3827.312700327269,3868.423226390294,3909.704957715744,3951.157495837971,3992.780445772349,4034.573415962834,4076.536018230648,4118.667867724050,4160.968582869160,4203.437785321818,4246.075099920432,4288.880154639816,4331.852580545961,4374.992011751736,4418.298085373485,4461.770441488504,4505.408723093361,4549.212576063055,4593.181649110980,4637.315593749676,4681.614064252349,4726.076717615142,4770.703213520135,4815.493214299054,4860.446384897675,4905.562392840909,4950.840908198545,4996.281603551630,5041.884153959494,5087.648236927371,5133.573532374628,5179.659722603578,5225.906492268863,5272.313528347389,5318.880520108815,5365.607159086563,5412.493139049352,5459.538155973243,5506.741908014168,5554.104095480955,5601.624420808817,5649.302588533310,5697.138305264736,5745.131279662996,5793.281222412867,5841.587846199706,5890.050865685565,5938.669997485708,5987.444960145534,6036.375474117870,6085.461261740653,6134.702047214979,6184.097556583517,6233.647517709273,6283.351660254702,6333.209715661160,6383.221417128693,6433.386499596144,6483.704699721581,6534.175755863042,6584.799408059578,6635.575398012603,6686.503469067529,6737.583366195699,6788.814835976588,6840.197626580299,6891.731487750303,6943.416170786471,6995.251428528350,7047.237015338692,7099.372687087244,7151.658201134776,7204.093316317348,7256.677792930807,7309.411392715536,7362.293878841399,7415.325015892938,7468.504569854757,7521.832308097153,7575.307999361927,7628.931413748417,7682.702322699730,7736.620498989168,7790.685716706857,7844.897751246550,7899.256379292639,7953.761378807334,8008.412529018027,8063.209610404841,8118.152404688335,8173.240694817405,8228.474264957325,8283.852900477976,8339.376387942222,8395.044515094454,8450.857070849281,8506.813845280385,8562.914629609517],
"averages": [0.2474039592545229,0.4794255386042030,0.6816387600233342,0.8414709848078965,0.9489846193555862,0.9974949866040544,0.9839859468739369,0.9092974268256817,0.7780731968879213,0.5984721441039564,0.3816609920523317,0.1411200080598672,-0.1081951345301084,-0.3507832276896198,-0.5715613187423438,-0.7568024953079282,-0.8949893582285835,-0.9775301176650970,-0.9992927889753780,-0.9589242746631385,-0.8589344934265921,-0.7055403255703919,-0.5082790774992583,-0.2794154981989259,-0.03317921654755682,0.2151199880878155,0.4500440737806176,0.6569865987187891,0.8230808790115054,0.9379999767747389,0.9945987791111761,0.9893582466233818,0.9226042102393402,0.7984871126234903,0.6247239537541924,0.4121184852417566,0.1738894853804336,-0.07515112046180931,-0.3195191936222736,-0.5440211108893699,-0.7346984304047954,-0.8796957599716700,-0.9699978679206785,-0.9999902065507035,-0.9678079975112615,-0.8754521746884285,-0.7286649758271700,-0.5365729180004349,-0.3111193549811273,-0.06632189735120068,0.1825991346311340,0.4201670368266409,0.6316109877182386,0.8037844265516210,0.9259824428086272,0.9906073556948704,0.9936411011327626,0.9348950555246830,0.8180217634546941,0.6502878401571169,0.4421221685765394,0.2064674819377966,-0.04202435271884079,-0.2879033166650653,-0.5158818468181092,-0.7117853423691232,-0.8634334728079056,-0.9613974918795568,-0.9995864713592172,-0.9756260054681576,-0.8910058399248534,-0.7509872467716761,-0.5642759039618552,-0.3424806184696125,-0.09939154689884817,0.1498772096629523,0.3898273272463786,0.6055398697196009,0.7836028759783553,0.9129452507276277,0.9855251115651197,0.9968297942787993,0.9461564284508708,0.8366556385360560,0.6751356532928010,0.4716390030941962,0.2388181240295827,-0.008851309290403876,-0.2559704110693330,-0.4871745124605095,-0.6880884622582969,-0.8462204041751706,-0.9517384599623535,-0.9980820279793963,-0.9823696896284233,-0.9055783620066239,-0.7724825579327704,-0.5913575298651244,-0.3734647547841147,-0.1323517500977730,0.1169902453743640,0.3590583540221682,0.5788019532877502,0.7625584504796028,0.8989027566124672,0.9793576431039170,0.9989208147888241,0.9563759284045030,0.8543682189235223,0.6992400316550977,0.5006364859324151,0.2709057883078690,0.02433148087720228,-0.2237556401867964,-0.4579307192868114,-0.6636338842129675,-0.8280755084772445,-0.9410314083429536,-0.9954785330494551,-0.9880316240928618,-0.9191536942035773,-0.7931272394572851,-0.6177879744108959,-0.4040376453230650,-0.1651662123735790,0.08397445569174683,0.3278939988258263,0.5514266812416906,0.7406743234096740,0.8838704235458307,0.9721117417027341,0.9999118601072672,0.9655423020447212,0.8711400001691764,0.7225744323811351,0.5290826861200238,0.3026951410063154,0.05748747810492456,-0.1912944777489552,-0.4281826694961510,-0.6384485367231334,-0.8090187662119064,-0.9292881272363391,-0.9917788534431158,-0.9926055741456128,-0.9317168878547055,-0.8128985581744033,-0.6435381333569994,-0.4341656243337532,-0.1977987996364623,0.05086619639930669,0.2963685787093853,0.5234441982331981,0.7179745927716441,0.8678648045849774,0.9637953862840878,0.9998018389311463,0.9736454556949781,0.8869525137988248,0.7451131604793488,0.5569462797382114,0.3341511768484206,0.09058017221711874,-0.1586226688047090,-0.3979631205566710,-0.6125601529754698,-0.7890711619691400,-0.9165215479156338,-0.9869870631127111,-0.9960865031195940,-0.9432541088290760,-0.8317747426285983,-0.6685796515669709,-0.4638155159838274,-0.2302135780707541,0.01770192510541358,0.2645168083164075,0.4948853175526282,0.6944842546815092,0.8509035245341184,0.9544177345154423,0.9985908724117705,0.9806764664578257,0.9017883476488092,0.7668313971238714,0.5841965844132857,0.3652392575940560,0.1235731227452240,-0.1257761903592125,-0.3673053491341913,-0.5859972403145538,-0.7682546613236668,-0.9027457284726567,-0.9811084386030970,-0.9984705779426957,-0.9537526527594719,-0.8497350070535673,-0.6928849542336957,-0.4929546708933114,-0.2623748537039288,-0.01548183890318817,0.2323737616554845,0.4657814871984436,0.6702291758433747,0.8330052605366240,0.9439891127251193,0.9962802940213323,0.9866275920404853,0.9156311650396451,0.7877052269841179,0.6108035931029846,0.3959251501818342,0.1564299990546788,-0.09279121175730869,-0.3362431144491572,-0.5587890488516162,-0.7465921866472573,-0.8879758383376634,-0.9741494532413135,-0.9997551733586199,-0.9632009590319782,-0.8667595742607592,-0.7164272772437812,-0.5215510020869119,-0.2942472117115055,-0.04864855487508726,0.1999748334780175,0.4361647552478249,0.6452360650659840,0.8141897215084345,0.9325210045313215,0.9928726480845371,0.9914922792803779,0.9284657227653786,0.8077116645594764,0.6367380071391379,0.4261750644253074,0.1891146203508915,-0.05970405485516449,-0.3048106211022167,-0.5309655392553846,-0.7241075918674496,-0.8722281415753229,-0.9661177700083929,-0.9999388748177166,-0.9715886235161092,-0.8828296974172939,-0.7391806966492229,-0.5495730203270262,-0.3257955554145617]
}
```

I have no idea!

**Comment by jamiebramwell** [comment]
> Out of curiosity, do these summary files exist purely for automated correctness-testing, or do we expect developers/users to be opening them up to check that values are in the expected ranges? If a human needs to be able to read them, we might consider adopting a different format, since this one will become difficult to read without external tools when there are more than a few timesteps present.
> 
> For example, what was happening around time `t = 4000` given the data below (using 250 timesteps as an example):
> 
> ```json
> {
> "t": [0,1.000000000000000,3.116658318641999,6.060086981831167,9.713559075160374,14.00589309483133,18.88722050361819,24.31984404491886,30.27384469521906,36.72465422735958,43.65158322401658,51.03686666650779,58.86501289862737,67.12234146015290,75.79664425067244,84.87693041290635,94.35322990663045,104.2164393606721,114.4581990969513,125.0707936057052,136.0470699770248,147.3803702967770,159.0644750535941,171.0935553362856,183.4621321274756,196.1650413838438,209.1974038785143,222.5545989954727,236.2322418290065,250.2261630667030,264.5323912321827,279.1471369404186,294.0667788792409,309.2878512791569,324.8070326726881,340.6211357761038,356.7270983522955,373.1219749347801,389.8029293103777,406.7672276726990,424.0122323707647,441.5353961873142,459.3342570899883,477.4064334058845,495.7496193762067,514.3615810530437,533.2401525048697,552.3832323012845,571.7887802508943,591.4548143691753,611.3794080557088,631.5606874624061,651.9968290362816,672.6860572220446,693.6266423112744,714.8168984262685,736.2551816278217,757.9398881372271,779.8694526637094,802.0423468293192,824.4570776840420,847.1121863045313,870.0062464704530,893.1378634129552,916.5056726302433,940.1083387656658,963.9445545440939,988.0130397627258,1012.312540332757,1036.841827368635,1061.599696321880,1086.584966156684,1111.796478564704,1137.233097216674,1162.893707048614,1188.777213580604,1214.882542266208,1241.208637870787,1267.754463877061,1294.519001916383,1321.501251224308,1348.700228119120,1376.114965502084,1403.744512378252,1431.587933396755,1459.644308409545,1487.912732047658,1516.392313314090,1545.082175192460,1573.981454270668,1603.089300378819,1632.404876240710,1661.927357138230,1691.655930588073,1721.589796030154,1751.728164527221,1782.070258475112,1812.615311323201,1843.362567304550,1874.311281175353,1905.460717963246,1936.810152724107,1968.358870306968,2000.106165126698,2032.051340944132,2064.193710653313,2096.532596075574,2129.067327760163,2161.797244791152,2194.721694600367,2227.840032786103,2261.151622937396,2294.655836463627,2328.352052429254,2362.239657393475,2396.318045254623,2430.586617099133,2465.044781054878,2499.691952148749,2534.527552168288,2569.551009527243,2604.761759134894,2640.159242269028,2675.742906452409,2711.512205332639,2747.466598565279,2783.605551700128,2819.928536070531,2856.435028685629,2893.124512125449,2929.996474438723,2967.050409043367,3004.285814629507,3041.702195064986,3079.299059303268,3117.075921293646,3155.032299893702,3193.167718783922,3231.481706384427,3269.973795773719,3308.643524609407,3347.490435050835,3386.514073683564,3425.713991445632,3465.089743555567,3504.640889442073,3544.366992675352,3584.267620900009,3624.342345769494,3664.590742882042,3705.012391718046,3745.606875578854,3786.373781526917,3827.312700327269,3868.423226390294,3909.704957715744,3951.157495837971,3992.780445772349,4034.573415962834,4076.536018230648,4118.667867724050,4160.968582869160,4203.437785321818,4246.075099920432,4288.880154639816,4331.852580545961,4374.992011751736,4418.298085373485,4461.770441488504,4505.408723093361,4549.212576063055,4593.181649110980,4637.315593749676,4681.614064252349,4726.076717615142,4770.703213520135,4815.493214299054,4860.446384897675,4905.562392840909,4950.840908198545,4996.281603551630,5041.884153959494,5087.648236927371,5133.573532374628,5179.659722603578,5225.906492268863,5272.313528347389,5318.880520108815,5365.607159086563,5412.493139049352,5459.538155973243,5506.741908014168,5554.104095480955,5601.624420808817,5649.302588533310,5697.138305264736,5745.131279662996,5793.281222412867,5841.587846199706,5890.050865685565,5938.669997485708,5987.444960145534,6036.375474117870,6085.461261740653,6134.702047214979,6184.097556583517,6233.647517709273,6283.351660254702,6333.209715661160,6383.221417128693,6433.386499596144,6483.704699721581,6534.175755863042,6584.799408059578,6635.575398012603,6686.503469067529,6737.583366195699,6788.814835976588,6840.197626580299,6891.731487750303,6943.416170786471,6995.251428528350,7047.237015338692,7099.372687087244,7151.658201134776,7204.093316317348,7256.677792930807,7309.411392715536,7362.293878841399,7415.325015892938,7468.504569854757,7521.832308097153,7575.307999361927,7628.931413748417,7682.702322699730,7736.620498989168,7790.685716706857,7844.897751246550,7899.256379292639,7953.761378807334,8008.412529018027,8063.209610404841,8118.152404688335,8173.240694817405,8228.474264957325,8283.852900477976,8339.376387942222,8395.044515094454,8450.857070849281,8506.813845280385,8562.914629609517],
> "averages": [0.2474039592545229,0.4794255386042030,0.6816387600233342,0.8414709848078965,0.9489846193555862,0.9974949866040544,0.9839859468739369,0.9092974268256817,0.7780731968879213,0.5984721441039564,0.3816609920523317,0.1411200080598672,-0.1081951345301084,-0.3507832276896198,-0.5715613187423438,-0.7568024953079282,-0.8949893582285835,-0.9775301176650970,-0.9992927889753780,-0.9589242746631385,-0.8589344934265921,-0.7055403255703919,-0.5082790774992583,-0.2794154981989259,-0.03317921654755682,0.2151199880878155,0.4500440737806176,0.6569865987187891,0.8230808790115054,0.9379999767747389,0.9945987791111761,0.9893582466233818,0.9226042102393402,0.7984871126234903,0.6247239537541924,0.4121184852417566,0.1738894853804336,-0.07515112046180931,-0.3195191936222736,-0.5440211108893699,-0.7346984304047954,-0.8796957599716700,-0.9699978679206785,-0.9999902065507035,-0.9678079975112615,-0.8754521746884285,-0.7286649758271700,-0.5365729180004349,-0.3111193549811273,-0.06632189735120068,0.1825991346311340,0.4201670368266409,0.6316109877182386,0.8037844265516210,0.9259824428086272,0.9906073556948704,0.9936411011327626,0.9348950555246830,0.8180217634546941,0.6502878401571169,0.4421221685765394,0.2064674819377966,-0.04202435271884079,-0.2879033166650653,-0.5158818468181092,-0.7117853423691232,-0.8634334728079056,-0.9613974918795568,-0.9995864713592172,-0.9756260054681576,-0.8910058399248534,-0.7509872467716761,-0.5642759039618552,-0.3424806184696125,-0.09939154689884817,0.1498772096629523,0.3898273272463786,0.6055398697196009,0.7836028759783553,0.9129452507276277,0.9855251115651197,0.9968297942787993,0.9461564284508708,0.8366556385360560,0.6751356532928010,0.4716390030941962,0.2388181240295827,-0.008851309290403876,-0.2559704110693330,-0.4871745124605095,-0.6880884622582969,-0.8462204041751706,-0.9517384599623535,-0.9980820279793963,-0.9823696896284233,-0.9055783620066239,-0.7724825579327704,-0.5913575298651244,-0.3734647547841147,-0.1323517500977730,0.1169902453743640,0.3590583540221682,0.5788019532877502,0.7625584504796028,0.8989027566124672,0.9793576431039170,0.9989208147888241,0.9563759284045030,0.8543682189235223,0.6992400316550977,0.5006364859324151,0.2709057883078690,0.02433148087720228,-0.2237556401867964,-0.4579307192868114,-0.6636338842129675,-0.8280755084772445,-0.9410314083429536,-0.9954785330494551,-0.9880316240928618,-0.9191536942035773,-0.7931272394572851,-0.6177879744108959,-0.4040376453230650,-0.1651662123735790,0.08397445569174683,0.3278939988258263,0.5514266812416906,0.7406743234096740,0.8838704235458307,0.9721117417027341,0.9999118601072672,0.9655423020447212,0.8711400001691764,0.7225744323811351,0.5290826861200238,0.3026951410063154,0.05748747810492456,-0.1912944777489552,-0.4281826694961510,-0.6384485367231334,-0.8090187662119064,-0.9292881272363391,-0.9917788534431158,-0.9926055741456128,-0.9317168878547055,-0.8128985581744033,-0.6435381333569994,-0.4341656243337532,-0.1977987996364623,0.05086619639930669,0.2963685787093853,0.5234441982331981,0.7179745927716441,0.8678648045849774,0.9637953862840878,0.9998018389311463,0.9736454556949781,0.8869525137988248,0.7451131604793488,0.5569462797382114,0.3341511768484206,0.09058017221711874,-0.1586226688047090,-0.3979631205566710,-0.6125601529754698,-0.7890711619691400,-0.9165215479156338,-0.9869870631127111,-0.9960865031195940,-0.9432541088290760,-0.8317747426285983,-0.6685796515669709,-0.4638155159838274,-0.2302135780707541,0.01770192510541358,0.2645168083164075,0.4948853175526282,0.6944842546815092,0.8509035245341184,0.9544177345154423,0.9985908724117705,0.9806764664578257,0.9017883476488092,0.7668313971238714,0.5841965844132857,0.3652392575940560,0.1235731227452240,-0.1257761903592125,-0.3673053491341913,-0.5859972403145538,-0.7682546613236668,-0.9027457284726567,-0.9811084386030970,-0.9984705779426957,-0.9537526527594719,-0.8497350070535673,-0.6928849542336957,-0.4929546708933114,-0.2623748537039288,-0.01548183890318817,0.2323737616554845,0.4657814871984436,0.6702291758433747,0.8330052605366240,0.9439891127251193,0.9962802940213323,0.9866275920404853,0.9156311650396451,0.7877052269841179,0.6108035931029846,0.3959251501818342,0.1564299990546788,-0.09279121175730869,-0.3362431144491572,-0.5587890488516162,-0.7465921866472573,-0.8879758383376634,-0.9741494532413135,-0.9997551733586199,-0.9632009590319782,-0.8667595742607592,-0.7164272772437812,-0.5215510020869119,-0.2942472117115055,-0.04864855487508726,0.1999748334780175,0.4361647552478249,0.6452360650659840,0.8141897215084345,0.9325210045313215,0.9928726480845371,0.9914922792803779,0.9284657227653786,0.8077116645594764,0.6367380071391379,0.4261750644253074,0.1891146203508915,-0.05970405485516449,-0.3048106211022167,-0.5309655392553846,-0.7241075918674496,-0.8722281415753229,-0.9661177700083929,-0.9999388748177166,-0.9715886235161092,-0.8828296974172939,-0.7391806966492229,-0.5495730203270262,-0.3257955554145617]
> }
> ```
> 
> I have no idea!

In my mind, this is more for integration testing.

File: scripts/testing/tolerance_test.py (hunk 3 above)
Comment (white238) [reply scripts/testing/tolerance_test.py]: went with 'stats'

**Pull request merged**